### DISCUSSION TITLE
20231116 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#66] [Meetings]
Time: 8:30 UTC (10:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@WarmBeer) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
Hi all (@da2ce7, @cgbosse, @mario-nt, @grmbyrn, @WarmBeer) for the next meeting I would like to add a topic for discussion. I would like to share my view on prioritising tasks and how I try to contribute.

### How to prioritize

1. Security bugs
2. Fix pipeline
3. Take care of the Community 
     3.1 Reply to new issues
     3.1 Reply comments
     3.2 Review PRs
     3.3 Merge PRs
4. Take care of the Team
     4.1 Reply team members waiting for feedback, review, …
     4.2 Help team members
5. Bugs
6. Keep things clean:
     6.1 Fix broken things
     6.2 Implement small issues (those taking less than 1 day)
         6.2.1 Update dependencies
         6.2.2 …
     6.3 Review and close stale issues
7. Outcome mindset
     7.1 Deliver things before starting new ones. “Good code is worthless in Git” (the sentence is not mine).
8. New features

In my daily routine, I always check that there is no higher-priority task.

### How to develop (values/principles/good practices)

- We all are maintainers. Keep an eye on everything, not only your current low-level task.
- Be part of the community. GitHub is not a TODO list.
     - Ask for feedback
     - Give feedback
     - Write good issues/comments
- Sostenible work
     - Don’t leave unfinished things
     - For big changes, make parallel changes. Be always production-ready
     - Don’t break things in a way that takes too much effort to restore them
     - Parallel and small changes. 
           - Introduce changes progressively.
           - They are easier to rollback
           - They are easier to integrate (merge).

# Resources

I think some companies are doing a great job explicitly defining their values/principles/practices, like for example, Audiense. See https://github.com/AudienseCo/engineering-principles.

Audiense Values

- 🏋️‍♀️ Strive for awesomeness
- ✋ Take ownership
- 🤜 🤛 Best as a team
- ✅ Do what I say I will do in the right way
- 🤩 Pursuit of happiness
- 👂 Everyone's voice is valued
- 🪟 Strive for a culture of transparency

Lean (Software/Product) Development Resources:

- https://www.linkedin.com/posts/eferro_lean-softwareproduct-development-resources-activity-7128692853869850624-ZKd7?utm_source=share&utm_medium=member_desktop

**Comment by mario-nt** [comment]
### GUI

- Rebased https://github.com/torrust/torrust-index-gui/pull/297
- Commented on https://github.com/torrust/torrust-index-gui/discussions/336

### INDEX

- Worked on https://github.com/torrust/torrust-index/issues/300

### Training/Research

- Ownership and borrowind in Rust
- Optionals, Results and error handling in Rust
- Structs in Rust
- V1 vs V2 vs Hybrid torrent files
- SHA1, SHA2/256
- Running Cypress tests in parallel

**Comment by josecelano** [comment]
I've been working on [reestablishing](https://github.com/torrust/torrust-index/issues/342) and [overhauling](https://github.com/torrust/torrust-index/issues/384) E2E tests.

### Torrust Index

PRs:

- Merged: [#[378](https://github.com/torrust/torrust-index/pull/378)] [Reestablish E2E Tests](https://github.com/torrust/torrust-index/pull/378)
- Merged: [#[383](https://github.com/torrust/torrust-index/pull/383)] [Reestablish E2E Tests (reopened)](https://github.com/torrust/torrust-index/pull/383)
- Merged: [#[381](https://github.com/torrust/torrust-index/pull/381)] [Use normal variables instead of secrets for user and repo on DockerHub](https://github.com/torrust/torrust-index/pull/381)
- Merged: [#[385](https://github.com/torrust/torrust-index/pull/385)] [Rename variables to follow conventions](https://github.com/torrust/torrust-index/pull/385)
- Merged: [#[387](https://github.com/torrust/torrust-index/pull/387)] [Remove deprecated env vars from container configuration](https://github.com/torrust/torrust-index/pull/387)
- Merged: [#[393](https://github.com/torrust/torrust-index/pull/393)] [Refactor: extract duplicate function wait_for_container_to_be_healthy](https://github.com/torrust/torrust-index/pull/393)

### Torrust Index GUI

PRs:

- Merged: [#[343](https://github.com/torrust/torrust-index-gui/pull/343)] [Rename index backend to index](https://github.com/torrust/torrust-index-gui/pull/343)
- Merged: [#[348](https://github.com/torrust/torrust-index-gui/pull/348)] [refactor: rename Index env vars to follow changes in the Index](https://github.com/torrust/torrust-index-gui/pull/348)

### Next

Continue working on [overhauling](https://github.com/torrust/torrust-index/issues/384) E2E tests and containers configuration in general.

**Comment by grmbyrn** [comment]
## Torrust Website
* Add contributor page with link on home page. Contributor page displays contributors with each linking to individual page with info on that contributor. Each post by that contributor is also then linked to their page.
* Add copy code button to Jose's most recent blog post.

## Secure Git Guide
* Find MegaLinter resources to help understand how to make it pass so site can be changed.

## Shoelace
* Research and learn how to implement web component library into a project.

## Meetups
* Find relevant meetups to network with other developers and let them know about Nautilus - found JS Barcelona meetup 22/11.

## Svelte/SK blog
* Continue writing and adding blogs on Svelte and SvelteKit.

## Python
* Continue learning about Python - classes.

**Comment by mickvandijke** [comment]
### Tracker
* Extracted the torrents map from the tracker struct into a new torrent repository struct. And added older torrent repository struct implementations for benchmarking purposes.
* Refactored the locks to sync in production code and all tests.
* Made a binary for benchmarking the torrent repository, especially the lock contention.
